isthmus/src/main/java/io/substrait/isthmus/SqlConverterBase.java

```diff
@@ -103,6 +110,39 @@ Pair<SqlValidator, CalciteCatalogReader> registerCreateTables(List<String> table
     return Pair.of(validator, catalogReader);
   }
 
+  Result registerCreateTablesForExtendedExpression(List<String> tables) throws SqlParseException {
+    Map<String, RelDataType> nameToTypeMap = new LinkedHashMap<>();
+    Map<String, RexNode> nameToNodeMap = new HashMap<>();
+    CalciteSchema rootSchema = CalciteSchema.createRootSchema(false);
+    CalciteCatalogReader catalogReader =
+        new CalciteCatalogReader(rootSchema, List.of(), factory, config);
+    SqlValidator validator = Validator.create(factory, catalogReader, SqlValidator.Config.DEFAULT);
+    if (tables != null) {
+      for (String tableDef : tables) {
+        List<DefinedTable> tList = parseCreateTable(factory, validator, tableDef);
+        for (DefinedTable t : tList) {
+          rootSchema.add(t.getName(), t);
+          for (RelDataTypeField field : t.type.getFieldList()) {
+            nameToTypeMap.put(
+                field.getName(), field.getType()); // to validate the sql expression tree
+            nameToNodeMap.put(
```

Comment:
Does this handle multiple tables containing a field with the same name?

Reply:
This scenario is not supported. It will be updated with the last filed type declared. Calcite API is not offering an option to mark field per this table if it is duplicated.

````
  /**
   * Converts an expression from {@link SqlNode} to {@link RexNode} format,
   * mapping identifier references to predefined expressions.
   *
   * @param node          Expression to translate
   * @param nameToNodeMap map from String to {@link RexNode}; when an
   *                      {@link SqlIdentifier} is encountered, it is used as a
   *                      key and translated to the corresponding value from
   *                      this map
   * @return Converted expression
   */
  public RexNode convertExpression(
      SqlNode node,
      Map<String, RexNode> nameToNodeMap) {
````

Reply:
If Calcite can't handle this, then we should throw an exception if we try to build either of those maps with duplicate values. I think it's better to explode loudly than to have a last-write wins style validator.

Reply:
One thing to check, does it work with qualified paths? Like if you have:
```sql
CREATE TABLE foo(a INT);
```
and you put the fully qualified identifier `foo.a` as the map key, can the expression `a` find it?

Then going further if you had
```sql
CREATE TABLE foo(a INT);
CREATE TABLE bar(a INT);
```
does Calcite reject the expression `a` as being ambiguous, but accept `foo.a`?

Reply:
I will add that validation and also test the last scenario you mentioned.

Reply:
Hi Team,

I am trying to implement this, but I am seeing limitations on the current Calcite implementation of SQL Expressions for qualified paths. There are two issues I would like to register:

1. Is needed to add support for qualified path at `SqlNode validSQLNode = validator.validateParameterizedExpression(sqlNode, nameToTypeMap);`
2. Is also needed to support for qualified path at `converter.convertExpression(validSQLNode, nameToNodeMap);`

Reply:
Just filled (1) https://issues.apache.org/jira/browse/CALCITE-6195.

Reply:
Thanks for looking into this! If qualified paths aren't supported Calcite, then I don't think we should concern ourselves with them here. I think we still need to deal with ambiguous lookups, probably by rejecting them if there is no mechanism to specify one or the other with a qualified path.

Reply:
Just added a logic to handle illegal argument for ambiguous lookups and also added unit test to validate that.